src/main/java/baritone/utils/schematic/litematica/LitematicaBlockStateContainer.java

```diff
@@ -0,0 +1,129 @@
+package baritone.utils.schematic.litematica;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3i;
+
+public class LitematicaBlockStateContainer implements ILitematicaBlockStatePaletteResizer
```

Comment:
This is also taken from the decompiled Minecraft game source. Once again, the game class `net.minecraft.world.chunk.BlockStateContainer` is not final, so you should extend it and then implement the additional functionality desired.